content/guides/astro.md

```diff
@@ -17,16 +17,164 @@ comments: false
+
+    export default defineConfig({
+      server: {
+        port: 8080,
+        host: true
+      }
+    });
+    ```
+
+  {{< /tab >}}
+
+
+{{< /tabs >}}
+
+## Deploy Astro  from the Console
+
+To deploy your Astro project to Clever Cloud, you **create a new application**. 
+
+{{% steps %}}
+
+### Create the application
+
+In this step, you set up the following parameters:
+
+- Deployment (using Git or GitHub)
+- Application : Node.js or Static
+- Instance size and scalability options : Astro sites can typically be deployed using the **Nano** instance
+- Region
+- Dependencies, if needed
+
+### Inject environment variables
+
+For **Node.js**, no specific environment variable is needed to deploy Astro if you're using **npm**. If you're using **yarn** or **pnpm**, or deploying on a Static application, set the following environment variables:
+
+{{% /steps %}}
+
+#### Node.js variables
+
+{{< tabs items="pnpm,yarn" >}}
+  
+  {{< tab >}}
+
+  ```shell
+    CC_NODE_BUILD_TOOL="custom"
+    CC_PRE_BUILD_HOOK="npm install -g pnpm && pnpm install"
+    CC_CUSTOM_BUILD_TOOL="pnpm run astro telemetry disable && pnpm build"
+    CC_RUN_COMMAND="pnpm run preview"
+    ```
+
+  {{< /tab >}}
+
+  {{< tab >}}
+
+  ```shell
+      CC_NODE_BUILD_TOOL="yarn"
+      CC_PRE_BUILD_HOOK="yarn && yarn run astro telemetry disable && yarn build"
+      CC_RUN_COMMAND="yarn run preview"
+    ```
+
+  {{< /tab >}}
+
+{{< /tabs >}}
+
+#### Static application variables
+
+{{< tabs items="npm,pnpm,yarn" >}}
+  
+  {{< tab >}}
+
+  ```shell
+    CC_POST_BUILD_HOOK="npm run build"
+    CC_PRE_BUILD_HOOK="npm install && npm run astro telemetry disable"
+    CC_WEBROOT="/dist"
+  ```
+
+  {{< /tab >}}
+
+  {{< tab >}}
+
+  ```shell
+    CC_POST_BUILD_HOOK="pnpm build"
+    CC_PRE_BUILD_HOOK="npm install -g pnpm && pnpm install && pnpm run astro telemetry disable"
+    CC_WEBROOT="/dist"
+  ```
+
+  {{< /tab >}}
+
+  {{< tab >}}
+
+  ```shell
+    CC_POST_BUILD_HOOK="yarn build"
+    CC_PRE_BUILD_HOOK="yarn && yarn run astro telemetry disable"
+    CC_WEBROOT="/dist"
+  ```
+
+  {{< /tab >}}
+
+{{< /tabs >}}
+
+### Deploy
+
+If you're deploying from **GitHub**, your deployment should start automatically. If you're using **Git**, copy the remote and push on the **master** branch.
+
+{{< callout emoji="💡" >}}
+  To deploy from branches other than `master`, use `git push clever <branch>:master`. For example, if you want to deploy your local `main` branch without renaming it, use `git push clever main:master`.
+{{< /callout >}}
+
+### Serve the site from server
+
+If the app uses the [Node.js adapter](https://docs.astro.build/en/guides/integrations-guide/node/), it can start by either:
+
+- Injecting `CC_RUN_COMMAND=./dist/server/entry.mjs` in environment variables. `CC_RUN_COMMAND` variable always overrides start script in `package.json`.
+- Setting `./dist/server/entry.mjs` as start script in `package.json`. 
+
+## Deploy an Astro site using the CLI
+
```

Comment:
I don't think it the good place for this. This remark is linked to other and the "Port and Host" part at the beginning of the guide. 

By default, and in the project used here, Astro build a static website. It is served from a server and a Clever Cloud user can host it using a static runtime, or another runtime with a static web server (in Node, Rust, Python, whatever). We recommend the static runtime as it's made for this (and will soon evolve to better respond to such needs). In this case, there is no specific Astro config to make, no port to define. And we can take profit of the build cache to just keep built website in it.

In the Astro config, the user can ask for the server to render all or some pages on-demand (SSR mode). `static` output should be replaced by `server` (dynamic website with some pre-rendered pages if asked) or `hybrid` (static website with some dynamic pages if asked).

In this case, if the Node adapter is installed and used, the server can be served with `node ./dist/server/entry.mjs` or using `./dist/server/entry.mjs` as a start script defined in `package.json`. But there are no case where `astro dev` or `astro preview` should be used to serve a website. It's only for dev/preview purposes.

All of this can be done on Clever Cloud through CLI or Console. 

So I think this part (and the host/port) should be in a SSR dedicated section after the static (simpler) one, explaining what are the differences/benefits of SSR (server or hybrid), the Node adapter, and what should be changed on the Astro config and env vars to use it. 

For pnpm/yarn, I think this make things more complex to understand when a user just want to deploy the provided example, and maybe should be detailed in a dedicated section. I let you decide.

Reply:
My bad, I didn't take into account that a default Astro project would be a static website and therefore can be deploy on Clever Cloud using static runtime or runtime with a static web server . Taking this in consideration and after going through other documentation, I agree that there should be a specific section for server rendering. So regarding this doc, maybe : first the most basic instructions for static Astro project, then specific instructions and explanations for SSR with the Port and host section, etc.

Otherwise I agree about the non usage of `astro dev` and `astro preview` in a deployment environment. 

About pnpm/yarn, I didn't feel like it was adding complexity, it is ok for me if it is not detailed in a specific section.

Reply:
Oops I resolved the wrong discussion, yes @HeleneAmouzou this was the input I needed, thank you for taking the time to review it!

Reply:
You both raised valid points regarding the deployment and clarity of the guide. These are the following points that I addressed in commit `4f5edb84`:
- Create a SSR section that includes host and port configuration
- Differentiate prod vs dev environment

The point about pnpm/yarn could add complexity, but as @HeleneAmouzou stated, it probably won't be complex for developers using these package managers. Astro users, and JS frameworks users in general, are widely familiar with them, and seeing a modern package manager as an option aligns the doc with the dynamism and the options available in this ecosystem.